core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -4882,4 +4928,112 @@ private boolean isH2DB(Connection dbConnection) throws Exception {
+                                    + jdbcConnectionCircuitBreakerState);
+                        }
+                        throw new CircuitBreakerException("Error occurred while obtaining connection.", e);
+                    }
+                } else {
+                    throw new CircuitBreakerException(
+                            "JDBC Connection circuit breaker is in open state for " + circuitOpenDuration
+                                    + "ms and has not reach the threshold timeout: " + thresholdTimeoutInMilliseconds
+                                    + "ms, hence avoid establishing the connection for " + getMyDomainName() +
+                                    " domain");
+                }
+            case CIRCUIT_STATE_CLOSE:
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug("connection circuit breaker state: " + jdbcConnectionCircuitBreakerState
+                                + ", so trying to obtain the connection for " + getMyDomainName() + " domain");
+                    }
+                    dbConnection = getConnection();
+                    break;
+                } catch (Exception e) {
+                    log.error("Error occurred while obtaining JDBC connection for " + getMyDomainName() +
+                            " domain", e);
+                    log.error("Trying again to get connection.");
+                    try {
+                        dbConnection = getConnection();
+                        break;
+                    } catch (Exception e1) {
+                        log.error("Error occurred while obtaining connection for the second time for "
+                                + getMyDomainName() + "domain", e1);
+                        jdbcConnectionCircuitBreakerState = CIRCUIT_STATE_OPEN;
+                        thresholdStartTime = System.currentTimeMillis();
+                        throw new CircuitBreakerException("Error occurred while obtaining connection."
+                                + " circuit breaker state set to: " + jdbcConnectionCircuitBreakerState, e1);
+                    }
+                }
+            default:
+                throw new UserStoreException("Unknown JDBC connection circuit breaker state.");
+        }
+        return dbConnection;
+    }
+
+    public long getThresholdStartTime() {
+        return thresholdStartTime;
+    }
+
+    public String getCircuitBreakerState() {
+        return jdbcConnectionCircuitBreakerState;
+    }
+
+    public long getThresholdTimeoutInMilliseconds() {
+        return thresholdTimeoutInMilliseconds;
+    }
+
+    /**
+     * Verify Circuit Breaker state for user store.
+     *
+     * @return true if CircuitBreaker Open, false otherwise.
+     */
+    public Boolean isCircuitBreakerOpen() {
+
+        long circuitOpenDuration = System.currentTimeMillis() - this.getThresholdStartTime();
+
+        if (this.getCircuitBreakerState().equals(CIRCUIT_STATE_OPEN)
```

Comment:
`CIRCUIT_STATE_OPEN.equals(this.getCircuitBreakerState())`

Use `<constant>.equals(<variable>)` pattern
check other places as well